downloader/src/main/kotlin/VcsHost.kt

```diff
@@ -446,6 +447,15 @@ enum class VcsHost(
                 host.toRawDownloadUrlInternal(userOrOrg, project, vcsInfo)
             }.getOrNull()
         }
+
+        /**
+         * Return the project's name.
+         */
+        fun getProject(projectUrl: String): String? {
+            val host = values().find { it.isApplicable(projectUrl) }
+                ?: return GIT_PROJECT_NAME.find(projectUrl)?.groupValues?.getOrNull(1)
+            return projectUrl.toUri { host.getProjectInternal(it) }.getOrNull()
+        }
```

Comment:
That does not work for self-hosted packages (e.g. an on-premise Git server).

At first I added a new value to the VcsHost enumeration, but I was not happy as it involves a constant hostname, which cannot be provided in this case. Therefore I implemented an approach similar to VcsHost.parsUrl.

Reply:
Or did you mean this methods implementation? I could change it to something like:
`return VcsHost.fromUrl(projectUrl)?.getProject(projectUrl)
                ?: GIT_PROJECT_NAME.find(projectUrl)?.groupValues?.getOrNull(1);`

Reply:
> That does not work for self-hosted packages (e.g. an on-premise Git server).

Yes, you're right. Didn't think about that.
I'm fine with the implementation as-is, would you mind adding the handling of "generic VCS hosts" to your `getProject()` function? To avoid confusion with the existing `getProject()` method?

Reply:
I extended the documentation as suggested, please have another look.